### PR TITLE
refactor: use Fail template tag elsewhere

### PR DESCRIPTION
**Pull request by erights**
The portions of #6482 not covered by other PRs

See https://github.com/endojs/endo/pull/1334

### CODE REVIEW

File: packages/agoric-cli/src/publish.js [hunk 1]
```diff
@@ -198,7 +196,9 @@ export const makeHttpBundlePublisher = ({ jsonHttpCall, getAccessToken }) => {
       'object',
       X`Expected JSON object response body, got ${response}`,
     );
-    assert(response, X`Expected non-null response body, got ${response}`);
+    if (!response) {
+      throw Fail`Expected non-null response body, got ${response}`;
+    }
```
Comment (turadg) [inline packages/agoric-cli/src/publish.js:201]: why not the `||` style?
```suggestion
    response || Fail`Expected non-null response body, got ${response}`;
```

Is there a guideline for when to use `||` or `if` ?

Comment (erights) [reply packages/agoric-cli/src/publish.js:201]: The guideline: Use `||` unless lint complains ;)

If you could help me get lint to complain less, we could use `||` more often.

Comment (turadg) [reply packages/agoric-cli/src/publish.js:201]: Ah, due to https://github.com/microsoft/TypeScript/issues/51426

Ok. If/when tsc can infer the type narrowing we can transform all the `if (expr) {throw …`

I noticed the issue is documented in SDK and Endo, https://github.com/Agoric/agoric-sdk/blob/6e3bc14aa0a77d8bdfad72ca8f0d8893c1f80b5f/packages/assert/src/types.js#L320-L324
https://github.com/endojs/endo/blob/412bc5e2ca37eaf6856ea35d3113ce9da6046e18/packages/ses/src/error/types.js#L315-L318

but the jsdoc isn't reaching these uses. Filed as https://github.com/endojs/endo/issues/1392

**Review by turadg — APPROVED** [review]

File: packages/agoric-cli/src/publish.js (hunk 1 above)
Comment (erights) [reply packages/agoric-cli/src/publish.js:201]: Thanks for filing that!

My interpretation is that, with this bug filed, there's nothing more I need to do here. So resolving.

**Pull request merged**